analyses/DESeq_analysis/run-DESeq-analysis.R

```diff
@@ -0,0 +1,200 @@
+
+# Read arguments from terminal
+args <- commandArgs(trailingOnly = TRUE)
+
+# Load required libraries
+suppressPackageStartupMessages({
+  library(ggplot2)
+  library(DESeq2)
+})
+
+
+HIST_index <- args[1]   #assign first argument to Histology index
+GTEX_index <- args[2]   #assign second argument to GTEx index
+
+    
+
+
+
+#Load histology file
+hist <- read.delim("../data/v6/histologies.tsv", header=TRUE, sep = '\t')
+
+#Load expression counts data
+countData <- readRDS("../data/v6/gene-counts-rsem-expected_count-collapsed.rds")
+
+#Load expression TPM data
+TPMData <- readRDS("../data/v6/gene-expression-rsem-tpm-collapsed.rds")
+
+#Load EFO-MONDO map file
+EFO_MONDO <- read.delim("../data/v6/efo-mondo-map.tsv", header =T, stringsAsFactors = FALSE)
+
+#Load gene symbol-gene ID RMTL file
+ENSG_Hugo <- read.delim("../data/v6/ensg-hugo-rmtl-v1-mapping.tsv", header =T)
+
+# Subset Histology file for samples only found in the current the countData file (To ensure no discepancies cause errors later in the code)
+hist.filtered <- unique(hist[which(hist$Kids_First_Biospecimen_ID %in%  colnames(countData)),])
+
+#Subset countdata for data that are present in the hitstoly files (To ensure no discepancies cause errors later in the code)
+countData_filtered <- countData[,which(colnames(countData) %in% hist$Kids_First_Biospecimen_ID)]
+
+#Subset countadata for data that are present in the hitstoly files (To ensure no discepancies cause errors later in the code)
+TPMData_filtered <- TPMData[,which(colnames(TPMData) %in% hist$Kids_First_Biospecimen_ID)]
+
+#Save all the unique cancer histologies in a variable. These cancer histologies represent the patient data in the countsdata
+Cancer_Histology <- unique(hist.filtered$cancer_group)
+
+#Save all the GTEx tissue subgroups in a variable. These cancer histologies represent the GTEx RNDA data available in the countsdata
+Gtex_Tissue_subgroup <- sort(unique(hist.filtered$gtex_subgroup))
+
+#Save all the cohorts represented in the countsdata into a variable. Remove all 'NA's from the list. 
+#And paste cohort to cancer groep (eg GMKF_Neuroblastoma)
+Cancer_Histology_COHORT <- unique(
+  paste(hist.filtered$cohort[which(!is.na(hist.filtered$cancer_group))],
+        hist.filtered$cancer_group[which(!is.na(hist.filtered$cancer_group))],
+        sep="_")
+  )
+
+#Save all the histologies represented in the countsdata into a variable. 
+#Remove all 'NA's from the list. 
+#This will be the basis of all the data from each histology combined regardless of cohort (eg all_cohorts_Neuroblastoma)
+Cancer_Histology <- paste("all_cohorts",Cancer_Histology[which(!is.na(Cancer_Histology))],sep="_")
+
+
+#Save all the GTEx subgroups represented in the countsdata into a variable. Remove all 'NA's 
+Gtex_Tissue_subgroup <- Gtex_Tissue_subgroup[!is.na(Gtex_Tissue_subgroup)]
```

Comment:
you can remove this if above in line 56, you do a selection for GTEX cohort (there are no NAs in that cohort list) before selecting the subgroups into a list

Reply:
Done.